sos/cleaner/__init__.py

```diff
@@ -0,0 +1,552 @@
+                with open(hn_file, 'r') as host_file:
+                    hostname = host_file.readline().strip()
+                    parser.load_hostname_into_map(hostname)
+            else:
+                self.obfuscate_file(
+                    archive.get_file_path(parser.prep_map_file)
+                )
+
+    def obfuscate_mac_from_line(self, line):
+        """Parse a line for presence of a MAC address, and if any are found
+        perform in place substitutions
+        """
+        pass
+
+    def obfuscate_ip_from_line(self, line):
+        """Parse a line for presence of an IP address, and if any are found
+        perform in place substitutions
+        """
+        pass
+
+    def obfuscate_file(self, filename, short_name=None):
+        """Obfuscate and individual file, line by line.
+
+        Lines processed, even if no substitutions occur, are then written to a
+        temp file without our own tmpdir. Once the file has been completely
+        iterated through, if there have been substitutions then the temp file
+        overwrites the original file. If there are no substitutions, then the
+        original file is left in place.
+
+        Positional arguments:
+
+            :param filename str:        Filename relative to the extracted
+                                        archive root
+        """
+        if not filename:
+            # the requested file doesn't exist in the archive
+            return
+        self.log_debug("Obfuscating %s" % filename)
+        subs = 0
+        tfile = tempfile.NamedTemporaryFile(mode='w', dir=self.tmpdir)
+        with open(filename, 'r') as fname:
+            for line in fname:
+                if not line.strip() or line.startswith('#'):
+                    continue
+                try:
+                    line, count = self.obfuscate_line(line, short_name)
+                    subs += count
+                    tfile.write(line)
+                except Exception as err:
+                    self.log_debug("Unable to obfuscate %s: %s"
+                                   % (filename, err))
+        tfile.seek(0)
+        if subs:
+            shutil.copy(tfile.name, filename)
+        tfile.close()
+        return subs
+
+    def obfuscate_line(self, line, filename):
+        """Run a line through each of the obfuscation parsers, keeping a
+        cumulative total of substitutions done on that particular line.
+
+        Positional arguments:
+
+            :param line str:        The raw line as read from the file being
+                                    processed
+            :param filename str:    Filename the line was read from
+
+        Returns the fully obfuscated line and the number of substitutions made
+        """
+        count = 0
+        for parser in self.parsers:
+            if filename and any([
+                re.match(_s, filename) for _s in parser.skip_files
+            ]):
+                continue
```

Comment:
(When) will be `skip_files` used? (so far this is unused) Is it like for skipping binary files, or some other "safe" files that we know they dont collect any sensitive data (but an attempt to obfuscate might harm)?

Reply:
In my earlier versions, one of the parsers used this however I found it unneeded with another change, however I kept this in here to allow future parsers to easily identify files they need to skip individually.